src/jwe.c

```diff
@@ -3340,17 +3358,32 @@ int r_jwe_decrypt_payload(jwe_t * jwe) {
   size_t payload_enc_len = 0, unzip_len = 0;
   unsigned char tag[128];
   size_t tag_len = 0;
+  size_t ciphertext_b64_len;
   int cipher_cbc;
   struct _o_datum dat = {0, NULL}, dat_ciph = {0, NULL}, dat_tag = {0, NULL};
 
-  if (jwe != NULL && jwe->enc != R_JWA_ENC_UNKNOWN && !o_strnullempty((const char *)jwe->ciphertext_b64url) && !o_strnullempty((const char *)jwe->iv_b64url) && jwe->key != NULL && jwe->key_len && jwe->key_len == _r_get_key_size(jwe->enc)) {
+  if (jwe != NULL && jwe->enc != R_JWA_ENC_UNKNOWN && (ciphertext_b64_len = o_strlen((const char *)jwe->ciphertext_b64url)) != 0 && !o_strnullempty((const char *)jwe->iv_b64url) && jwe->key != NULL && jwe->key_len && jwe->key_len == _r_get_key_size(jwe->enc)) {
+    /* ensure payload_enc_buflen is a multiple of cipher_block_size
+     * if the cipher is a block-mode cipher
+     */
+    if (gnutls_is_block_cipher(_r_get_alg_from_enc(jwe->enc))) {
+      size_t ciphertext_decoded_len = (ciphertext_b64_len * 3) / 4;
+      unsigned cipher_block_size = (unsigned)gnutls_cipher_get_block_size(_r_get_alg_from_enc(jwe->enc));
+      if (ciphertext_decoded_len % cipher_block_size) {
+        /* The ciphertext length is not a multiple of block size.
+        * It can't possibly be valid */
+        y_log_message(Y_LOG_LEVEL_ERROR, "r_jwe_decrypt_payload - Invalid ciphertext length");
+        ret = RHN_ERROR_INVALID;
+        return ret;
```

Comment:
I prefer having one return statement in a function, at the end of it. You can add an `else` statement and surround the `// Decode iv and payload_b64` in it.

Reply:
done